server/src/main/java/no/fractal/util/StreamUtils.java

```diff
@@ -0,0 +1,108 @@
+
+        int remaining = byteSize;
+        int bufferSize = 4096;
+
+        byte[] buffer = new byte[bufferSize];
+        int bytesRead = -1;
+
+        while ((bytesRead = inputStream.read(buffer, 0, Math.min(remaining, bufferSize))) != -1) {
+            remaining -= bufferSize;
+
+            outputStream.write(buffer, 0, bytesRead);
+        }
+    }
+
+    // -- get numbers -- /
+    public static short readFirstShort(BufferedInputStream inputStream) throws IOException {
+        return ByteBuffer.wrap(inputStream.readNBytes(2)).getShort();
+    }
+
+    public static int readFirstInt(BufferedInputStream inputStream) throws IOException {
+        return ByteBuffer.wrap(inputStream.readNBytes(4)).getInt();
+    }
+
+    public static Long readFirstLong(BufferedInputStream inputStream) throws IOException {
+        return ByteBuffer.wrap(inputStream.readNBytes(8)).getLong();
+    }
+
+    public static Float readFirstFloat(BufferedInputStream inputStream) throws IOException {
+        return ByteBuffer.wrap(inputStream.readNBytes(4)).getFloat();
+    }
+
+    public static Double readFirstDouble(BufferedInputStream inputStream) throws IOException {
+        return ByteBuffer.wrap(inputStream.readNBytes(4)).getDouble();
+    }
+
+    // -- write numbers -- //
+    public static void writeShort(BufferedOutputStream outputStream, short val) throws IOException {
+        ByteBuffer buffer = ByteBuffer.allocate(2);
+        buffer.putShort(val);
+        outputStream.write(buffer.array());
+    }
+
+    public static void writeInt(BufferedOutputStream outputStream, int val) throws IOException {
+        ByteBuffer buffer = ByteBuffer.allocate(4);
+        buffer.putInt(val);
+        outputStream.write(buffer.array());
+    }
+
+    public static void writeToStream(BufferedOutputStream outputStream, Iterator<byte[]> iterator) throws IOException {
+        while (iterator.hasNext()) {
+            outputStream.write(iterator.next());
+        }
+    }
+
+    public static void writeFileToStream(BufferedOutputStream outputStream, File file) throws IOException {
+        writeFileToStream(outputStream, file, 4096);
+    }
+
+    public static void writeFileToStream(BufferedOutputStream outputStream, File file, int bufferSize) throws IOException {
+        BufferedInputStream inputStream = new BufferedInputStream(new FileInputStream(file));
+
+
+        byte[] buffer = new byte[bufferSize];
+        int bytesRead = -1;
+
+        while ((bytesRead = inputStream.read(buffer)) < 0) {
+            outputStream.write(buffer, 0, bytesRead);
+        }
+    }
+
+    public static void writeStringToStream(BufferedOutputStream outputStream, String string) throws IOException {
+        outputStream.write(string.getBytes());
+    }
+
+    public static void writeJsonToStream(BufferedOutputStream outputStream, JSONObject jsonObject) throws IOException {
+        writeStringToStream(outputStream, jsonObject.toString());
+
+
```

Comment:
Hvorfor trenger vi denne når du egentlig bare gjør den til string? kan bare bruke den over da?

Reply:
bare for lesbarhet så man lettere kan se nedover hva som leses til stømmen uten og måtte nilese etter en .tostring